### PR TITLE
[ready] `add`: kokkos impl 

### PR DESCRIPTION
**Pull request by fnrizzi**
#162  is a prerequisite for this, I can rebase once that is merged

### CODE REVIEW

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp [hunk 1]
```diff
@@ -22,19 +30,72 @@ struct LayoutMapper<std::experimental::layout_right> {
...
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==2, Kokkos::View<typename MDSpan::element_type**,
-                                                 typename LayoutMapper<typename MDSpan::layout_type>::type>>
-mdspan_to_view(const MDSpan& a) {
-    return Kokkos::View<typename MDSpan::element_type**,
-                        typename LayoutMapper<typename MDSpan::layout_type>::type>
-            (a.data(),a.extent(0),a.extent(1));
+std::enable_if_t<
+  MDSpan::rank()==1
+  && is_complex_v<typename MDSpan::element_type>,
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp:45]: This is incorrect for `const` and/or `volatile`-qualified `element_type`.  Please see my comments on PR #162.

**Review by mhoemmen — CHANGES_REQUESTED** [review]

**Comment by fnrizzi** [comment]
@mhoemmen this is ready IMO

**Review by mhoemmen — APPROVED** [review]
Looks great!  Thanks for the contribution and for your responses to feedback!

It looks like there's a merge conflict, so I can't merge this one, alas.

**Comment by fnrizzi** [comment]
> Looks great! Thanks for the contribution and for your responses to feedback!
> 
> It looks like there's a merge conflict, so I can't merge this one, alas.

conflict fixed via #172

**Pull request merged**